pkg/grpc/api.go

```diff
@@ -1377,13 +1377,45 @@ func (a *api) SubscribeConfigurationAlpha1(request *runtimev1pb.SubscribeConfigu
 		apiServerLogger.Debug(err)
 		return err
 	}
-
 	sort.Slice(request.Keys, func(i, j int) bool {
 		return request.Keys[i] < request.Keys[j]
 	})
 
+	subscribeKeys := request.Keys
+	unsubscribedKeys := make([]string, 0)
+
+	newCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// empty list means subscribing to all configuration keys
+	if len(subscribeKeys) == 0 {
+		getConfigurationReq := &runtimev1pb.GetConfigurationRequest{
+			StoreName: request.StoreName,
+			Keys:      []string{},
+			Metadata:  request.GetMetadata(),
+		}
+		resp, err2 := a.GetConfigurationAlpha1(newCtx, getConfigurationReq)
+		if err2 != nil {
+			err2 = status.Errorf(codes.Internal, fmt.Sprintf(messages.ErrConfigurationGet, request.Keys, request.StoreName, err2))
+			apiServerLogger.Debug(err2)
+			return err2
+		}
+
+		items := resp.GetItems()
+		for _, item := range items {
+			if _, ok := a.configurationSubscribe[fmt.Sprintf("%s||%s", request.StoreName, item.Key)]; !ok {
+				unsubscribedKeys = append(unsubscribedKeys, item.Key)
+			}
+		}
+	} else {
+		for _, k := range subscribeKeys {
+			if _, ok := a.configurationSubscribe[fmt.Sprintf("%s||%s", request.StoreName, k)]; !ok {
+				unsubscribedKeys = append(unsubscribedKeys, k)
+			}
+		}
+	}
+
 	req := &configuration.SubscribeRequest{
-		Keys:     request.Keys,
+		Keys:     unsubscribedKeys,
```

Comment:
Why is this called "unsubscribedKeys"? Is it being unsubscribed? This might just be a naming issue.

Reply:
@artursouza I have renamed the variable. Please review this.